src/Symfony/Component/DependencyInjection/Argument/ServiceLocatorArgument.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * Represents a service locator able to lazy load a given range of services.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ *
+ * @experimental in version 3.3
+ */
+class ServiceLocatorArgument implements ArgumentInterface
+{
+    private $values;
+
+    /**
+     * @param array $values An array of mixed entries indexed by identifier
```

Comment:
That's PSR-11 - return type of "get" is mixed for this reason - same as eg Pimple ;)
I see no reason to forbid non-objects - that won't provide any technical benefit (even if I won't use that much myself).